base/module/src/context/motis_http_req.cc

```diff
@@ -76,13 +77,24 @@ struct http_request_executor
       }
     }
 
+    url target_url;
+    try {
+      target_url = url(target);
+    } catch (std::invalid_argument const&) {
+      // not an absolute redirect, try relative redirect with absolute path
+      // (technically we'd also need to handle relative paths here)
+      target_url = url(fmt::format("{}://{}{}", request_url_.prot(),
+                                   request_url_.host(), target));
```

Comment:
Does this work if the redirect is e.g. to `foo`, or `./foo`, or `../foo/bar`?

Reply:
No, as mentioned redirects to relative paths are still not supported, that will continue to produce an exception as before. This is just patching up the bare minimum needed to make all GTFS-RT feeds currently in Transitous work.

Fixing this properly is more effort, but before doing that I'd suggest to explore using an existing HTTP/URL/etc implementation instead of continuing to do all this manually.

Reply:
Yes, 💯 for using an off-the-shelf HTTP client. (libcurl?)

Reply:
Using [Boost.URL](https://www.boost.org/doc/libs/1_82_0/libs/url/doc/html/url/ref/boost__urls__resolve.html) would probably be more robust and should handle these cases.

Reply:
Both, using libcurl as well as switching to Boost.URL would need to be done in the [`net` library](https://github.com/motis-project/net). Long term, switching to something off-the-shelf makes sense. But to enable MOTIS today to work with relative redirects, I'll merge it as it is.

Reply:
Oh, indeed, that is much better! Seeing a handwritten URL class here I didn't even think about that we do have Boost available already.